sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubProducer/EventHubProducerLiveTests.cs

```diff
@@ -181,5 +181,302 @@ await using (var producer = client.CreateProducer())
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    var producerOptions = new EventHubProducerOptions { PartitionId = null };
+
+                    await using (var producer = client.CreateProducer(producerOptions))
+                    {
+                        var events = new[] { new EventData(Encoding.UTF8.GetBytes("Will it work")) };
+                        Assert.That(async () => await producer.SendAsync(events), Throws.Nothing);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task ProducerCannotSendWhenClosed(bool sync)
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    var events = new[] { new EventData(Encoding.UTF8.GetBytes("Dummy event")) };
+                    Assert.That(async () => await producer.SendAsync(events), Throws.Nothing);
+
+                    if (sync)
+                    {
+                        producer.Close();
+                    }
+                    else
+                    {
+                        await producer.CloseAsync();
+                    }
+
+                    Assert.ThrowsAsync<ObjectDisposedException>(async () => await producer.SendAsync(events));
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProducerCannotSendToInvalidPartition()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    var events = new[] { new EventData(Encoding.UTF8.GetBytes("Lorem Ipsum")) };
+
+                    // Some invalid partition values. These will fail on the service side.
+                    var invalidPartitions = new[]
+                    {
```

Comment:
Can we please move these to `[TestCase]` arguments?   Embedding data in this way makes understanding a failure more difficult, and goes against the "test only one thing" guidance.

Reply:
Done.